pyface/util/testing.py

```diff
@@ -8,6 +8,21 @@
 #
 # Thanks for using Enthought open source!
 from functools import wraps
+import re
+from unittest import TestSuite
+
+
+def filter_tests(test_suite, exclusion_pattern):
+    filtered_test_suite = TestSuite()
+    for item in test_suite:
+        if isinstance(item, TestSuite):
+            filtered = filter_tests(item, exclusion_pattern)
+            filtered_test_suite.addTest(filtered)
+        else:
+            match = re.search(exclusion_pattern, item.id())
+            if match is None:
```

Comment:
It would be nice to be able to record excluded tests (perhaps by creating fake skipped tests with the same ID), as that would help guard against spurious matches, but this is OK.

Reply:
I implemented this, but the module of skipped tests usually has import errors so they are logged a bit differently:
```
pyface.ui.qt4.workbench.tests.test_workbench_window_layout (unittest.loader._FailedTest) ... skipped 'Test excluded via pattern qt'
test_get_column_count (pyface.ui.wx.grid.tests.test_composite_grid_model.CompositeGridModelTestCase) ... ok
```